lisa/sut_orchestrator/baremetal/readychecker.py

```diff
@@ -115,12 +109,19 @@ def is_ready(self, node: Node) -> bool:
         context = get_node_context(node)
         remote_node = cast(RemoteNode, node)
         remote_node.set_connection_info(
-            **fields_to_dict(
-                context.connection,
-                ["address", "port", "username", "password", "private_key_file"],
+            address=context.connection.address,
```

Comment:
Why change this part?

Reply:
set_connection_info() has both port and public_port parameter with default value at 22, where public_port value is used to connect with the node during later stage. fields_to_dict(context.connection,  ["address", "port", "username", "password", "private_key_file"]) only update the port value, instead of public_port value, which cause connection issue to the node that has access port value other than 22.

Reply:
I think you can set both public_port and port in runbook. The public_port/port is used to allow different ports for LISA/node and node/node communication.

Reply:
I tried to set both public_port and port to be same value other than 22, however, from my debug, context.connection structure contains port value only, fields_to_dict(context.connection, ["address", "port", "username", "password", "private_key_file"] pass this port value to set_connection_info() function, but the default public_port parameter of set_connection_info() is still 22. Unfortunately, set_connection_info() use public_port value instead of port value as the connection port from its implementation. If set_connection_info() is called, its input parameter of public_port needs to be updated in order to connect the node successfully.

Reply:
@LiliDeng why there is to set connection info? How about call node.set_connection_info below?

https://github.com/paulli2017/lisa/blob/709497ef75f8178ab1d730946958f9be1c5ef85c/lisa/sut_orchestrator/baremetal/platform_.py#L191-L192